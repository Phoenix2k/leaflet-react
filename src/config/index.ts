import { Config } from '../types';
import emotionTheme from './emotion';
import materialUIOptions from './material-ui';
import pages from './pages';

const config: Config = {
  a11yEnv: ['development'],
  emotion: emotionTheme,
  materialUI: materialUIOptions,
  pages
};

export default config;
