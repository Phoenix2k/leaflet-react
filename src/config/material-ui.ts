import { ThemeOptions } from '@material-ui/core/';

const materialUIOptions: ThemeOptions = {
  palette: {
    primary: {
      main: '#4caf50'
    },
    secondary: {
      main: '#2e7d32'
    }
  }
};

export default materialUIOptions;
