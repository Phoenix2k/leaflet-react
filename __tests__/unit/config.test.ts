/// <reference types="jest"/>

import config from '../../src/config';

describe('Config', () => {
  it('matches specification', () => {
    expect(Object.keys(config).length).toEqual(4);
    expect(Object.keys(config.pages).length).toBeGreaterThan(0);
    expect(Object.keys(config.emotion).length).toBeGreaterThan(0);
  });
  it('matches snapshot', () => {
    expect(config).toMatchSnapshot();
  });
});
