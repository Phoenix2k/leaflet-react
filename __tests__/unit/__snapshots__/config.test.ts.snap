// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config matches snapshot 1`] = `
Object {
  "a11yEnv": Array [
    "development",
  ],
  "emotion": Object {
    "classes": Object {
      "navLinkActive": "nav-link--active",
    },
    "colors": Object {
      "backgroundColor": "rgb(255,255,255)",
      "borderColor": "#ccc",
      "color": "black",
      "navLink": "blue",
      "navLinkActive": "rgb(0, 128, 0)",
    },
    "typography": Object {
      "sansSerif": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
    },
  },
  "materialUI": Object {
    "palette": Object {
      "primary": Object {
        "main": "#4caf50",
      },
      "secondary": Object {
        "main": "#2e7d32",
      },
    },
  },
  "pages": Object {
    "home": Object {
      "navLabel": "Home",
      "slug": "home",
      "title": "Leaflet ðŸŒ± React",
      "url": "/",
    },
    "map": Object {
      "navLabel": "Map",
      "slug": "map",
      "title": "Map",
      "url": "/map",
    },
  },
}
`;
